llconfig/__init__.py

```diff
@@ -71,17 +71,21 @@ def init(self, key: str, converter: Callable[[str], Any], default=None):
         Args:
             key: Case-sensitive directive name which is used everywhere (in env vars, in config files, in defaults).
             converter: Function, which is called when converting env variable value to Python.
-            default: Directive default value.
+            default: Directive default value. Converter is applied on it.
         """
         if key == self.config_files_env_var:
             raise KeyError('Conflict between directive name and `config_files_env_var` name.')
 
         _check_safe_env_name(key)
 
         self._loaded = False
-        self._default_layer[key] = default
+        self._default_layer[key] = converter(default)
```

Comment:
Please don't convert the default. The general concept is, that you know what you write (whether as a default or to the config file). The conversion is only necessary because of env variables.